File: src/main/java/MyServer.java
import java.io.*;
import java.net.*;

public class MyServer {
    public static void main(String[] args) {
        try {
            ServerSocket ss = new ServerSocket(8080);
            System.out.println(">> Server is On!");
            while (true) {
                Socket s = ss.accept();//establishes connection
                DataInputStream dis = new DataInputStream(s.getInputStream());
                String str = (String) dis.readUTF();
                System.out.println("message= " + str);
                if (str.equals("end")) {
                    ss.close();
                    System.out.println(">> Server is Off!");
                    break;
                }
            }
        } catch (Exception e) {
            System.out.println(e);
        }
    }
}  

File: src/main/scala/MyServer.scala
import java.io.*
import java.net.*

object MyServer {
  def main(args: Array[String]): Unit =
    try {
      val ss = new ServerSocket(8080)
      var loop_stat = true
      println(">> Server is On!")
      while (loop_stat) {
        val s = ss.accept() // establishes connection
        val dis = new DataInputStream(s.getInputStream)
        val str = dis.readUTF()
        println("message= " + str)
        if (str == "end") {
          ss.close()
          println(">> Server is Off!")
          loop_stat = false
        }
      }
    } catch {
      case e: Exception => println(e)
    }
}

